docs/API.md

```diff
@@ -0,0 +1,285 @@
+# Shipyard API
+
+Logically, the API has three parts to handle the three areas of functionality in Shipyard.
+
+1. Document Staging
+2. Action Handling 
+3. Airflow Monitoring 
+
+There are several standard error responses that should be handled appropriately by a client of this API:
+* 401 Authentication Error  
+If not authenticated
+* 403 Forbidden  
+If not authorized
+* 404 Not found  
+If a variable in the path doesn't reference a valid resource 
+* 503 Service Unavailable  
+Error if an upstream system cannot be reached (e.g. Deckhand for documents, Airflow for actions)
+
+---
+## Headers
+### Required
+
+* X-Auth-Token - The auth token to identify the invoking user.
+
+### Optional
+
+* X-Context-Marker - A context id that will be carried on all logs for this cleint-provided marker. This marker may only be a 36-character canonical representation of an UUID  (8-4-4-4-12)
+
+---
+## <a name="DocumentStagingAPI"></a> Document Staging API
+Shipyard will serve as the entrypoint for documents and secrets into a site. At any point in time, there will be two represented versions of documents in a site that are accessible via this API. 
+
+* The lastDeployed version, which represents the last version of documents that successfully completed deployment.
+* The staged version, which represents the document set that has been ingested by this API since the lastDeployed version.
+
+### /v1.0/configDocs  
+Represents the site configuration documents.  
+
+#### Payload Structure
+The yaml documents separated by --- and concluded by ...
```

Comment:
Minor nit: Is a YAML document conclusion going to be an absolute requirement? My understanding is its optional.